### ISSUE TITLE
[Feature Request] Support customizing requests sent by TanStack Query

### ISSUE THREAD

**Issue by ymc9** [zenstackhq/zenstack#541]
Right now the hooks code generated by "@zenstackhq/tanstack-query" plugin sends requests directly by calling `fetch`, and users don't have a way to do customizations like adding extra headers. It's designed to have some global configuration mechanism for customization.

One possible solution is to support passing in a custom `fetch` function from the context:

```ts
function App({ Component, pageProps: { session, ...pageProps } }: AppProps) {
    return (
        <QueryClientProvider client={queryClient}>
            <ZenStackHooksProvider value={{ endpoint: '/api/model', fetch: myFetch }}>
                <AppContent />
            </ZenStackHooksProvider>
        </QueryClientProvider>
    );
}
```

Internally, the generated code checks if there's a `fetch` from the context, and if so, use it instead.

**Comment by ymc9** [comment]
Requested by kiwicrog (on Discord) and @chemitaxis.

**Comment by ymc9** [comment]
Hi @chemitaxis , thanks for proposing to work on this!

A bit of explanation about the code structure of "tanstack-query". The files under "res" folder are "template code" that're copied over to the generation target folder when the plugin runs. Some of the template code is framework specific and resides in separate folders: "res/react" and "res/svelte". 

The HTTP requests are made in the `fetcher<R>` function from "shared.ts". If we can pass in a custom `fetch` function here, the problem can be solved. What do you think?

**Comment by ymc9** [comment]
Fixed by #556
